pydoctor/epydoc2stan.py

```diff
@@ -403,17 +413,31 @@ def set_param_types_from_annotations(
             if not _is_none_literal(ann_ret):
                 self.return_desc = FieldDesc(type=ret_type)
 
-    def handle_return(self, field: Field) -> None:
+    @staticmethod
+    def _report_unexpected_argument(field:Field) -> None:
         if field.arg is not None:
             field.report('Unexpected argument in %s field' % (field.tag,))
+
+    def handle_return(self, field: Field) -> None:
+        self._report_unexpected_argument(field)
         if not self.return_desc:
             self.return_desc = FieldDesc()
         self.return_desc.body = field.format()
     handle_returns = handle_return
 
+    # TODO support types for yields section
+    # Currently, type will be shown as it's included in napoleon processed strings
+    # but not it's not "the right place". https://github.com/sphinx-doc/sphinx/issues/8004
+    # We could add support for ':ytype:' and use it inside pydoctor.napoleon. 
```

Comment:
I suggest we leave this as is for now.

Reply:
Maybe it's better to have it the right way from the start. So we should introduce this ``:ytype:`` field, I think.